Editorial review: Document CSS multicol column wrapping (#42934)
mdn/content: files/en-us/web/css/guides/multicol_layout/handling_overflow/index.md

## More columns than will fit

How overflowing columns are handled depends on whether the media context is fragmented, such as print, or is continuous, such as a web page. It also depends on whether {{cssxref("column-height")}} and {{cssxref("column-wrap")}} are being used to enforce [column wrapping](#using_column_wrapping_for_multicol).

> In fragmented media, after a fragment (for example, a page) is filled with columns, the columns will move to a new page and fill that up with columns. In continuous media, columns will overflow in the inline direction. On the web, this means that you will get a horizontal scrollbar.

Author: I've made this change, and I've also added a note to mention the height restriction thing.

How overflowing columns are handled depends on whether the media context is fragmented, such as print, or is continuous, such as a web page.

- In fragmented media, after a fragment (for example, a page) is filled with columns, the columns will move to a new page and fill that up with columns.

> - In continuous media, the default behavior is that columns overflow in the inline direction. On the web, this means that you will get a horizontal scrollbar. This behavior can be overidden by using {{cssxref("column-height")}} and {{cssxref("column-wrap")}} to enforce [column wrapping](#using_column_wrapping_for_multicol).

Reviewer: ```suggestion
- In continuous media, if a height is set on a multi-column container, columns overflow in the inline direction by default. On the web, this means that you will get a horizontal scrollbar. This behavior can be overridden by using {{cssxref("column-height")}} and {{cssxref("column-wrap")}} to enforce [column wrapping](#using_column_wrapping_for_multicol).
```

> [!NOTE]
> It is worth mentioning that overflow in the inline direction only occurs when you've restricted the height of the multicol container.

> The example below shows the default continuous media overflow behavior. The multicol container has a set {{CSSXref("height")}} and there is more text than space to create columns; therefore, we get columns created outside of the container.

Reviewer: ```suggestion
This example shows the default continuous media overflow behavior. The multicol container has a set {{CSSXref("height")}} and there is more text than space to create columns; therefore, we get columns created outside of the container.
```
i think we're still avoiding "below" and "above" on mdn

  columns, and placed into multiple rows.
</p>

<p>
  The <code>column-height</code> and <code>column-wrap</code> properties were
  introduced in
  <a href="https://drafts.csswg.org/css-multicol-2/"
    >CSS Multi-column Layout Module Level 2</a
  >.
</p>
```

We give our content some styles. Most notably, we set the `<body>` element's {{cssxref("column-count")}} to `2`, and its `column-height` to `95vh` so that each row of columns fills up the viewport. We don't need to explicitly set `column-wrap` to `wrap`: when `column-height` is set to a {{cssxref("&lt;length>")}} value, the initial value of `column-wrap` (`auto`) resolves to `wrap`, which is usually the behavior you'll want.

```css hidden
html {
  font-family: sans-serif;
}

body {
  width: 60%;
  margin: 0 auto;
}

p:first-of-type {
  margin-top: 0;
}
```

```css
body {
  font-size: 1.3em;
  line-height: 1.5;
  column-count: 2;
  column-height: 95vh;
}
```

```css hidden
@supports not (column-height: 5em) {
  body::before {
    content: "Your browser does not support the 'column-height' property.";
    background-color: wheat;
    position: fixed;
    inset: 40% 0;

>     height: fit-content;

Reviewer: i think that is the default for fixed

  columns to wrap onto multiple rows. The text is equally distributed over the
  columns, and placed into multiple rows.
</p>

<p>
  The <code>column-height</code> and <code>column-wrap</code> properties were
  introduced in
  <a href="https://drafts.csswg.org/css-multicol-2/"
    >CSS Multi-column Layout Module Level 2</a
  >.
</p>
```

We give our content some styles. Most notably, we set the `<body>` element's `column-count` to `2`, and its `column-height` to `95vh` so that each row of columns fills up the viewport. We don't need to explicitly set `column-wrap` to `wrap`: when `column-height` is set to a {{cssxref("&lt;length>")}} value, the initial value of `column-wrap` (`auto`) resolves to `wrap`, which is usually the behavior you'll want.

```css hidden
html {
  font-family: sans-serif;
}

body {
  width: 60%;
  margin: 0 auto;
}

p:first-of-type {
  margin-top: 0;
}
```

```css
body {
  font-size: 1.3em;
  line-height: 1.5;
  column-count: 2;
  column-height: 95vh;
}
```

```css hidden
@supports not (column-height: 5em) {
  body::before {
    content: "Your browser does not support the 'column-height' property.";

>     color: black;

Reviewer: ```suggestion
```

    background-color: #ffcd33;

>     display: block;

Reviewer: ```suggestion
```

Author: Updated. I've made the changes you've requested in this comment and the ones below, plus some others, in all `@supports` blocks in this PR, to make them more consistent with the other banners across MDN.
